src/io/iter_image_recordio_2.cc

```diff
@@ -285,9 +285,14 @@ inline bool ImageRecordIOParser2<DType>::ParseNext(DataBatch *out) {
     shape_vec.push_back(param_.label_width);
     TShape label_shape(shape_vec.begin(), shape_vec.end());
 
-    out->data.at(0) = NDArray(data_shape, Context::CPU(0), false,
+    auto ctx = Context::CPU(0);
+    auto dev_id = param_.device_id;
```

Comment:
What if a device id is invalid or larger than the number of GPUs in the instance?

Reply:
It will throw error in cuda: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading CUDA: invalid device ordinal